text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,329 @@
+* CDK Assets (CDK File Assets/CDK Docker Assets)
+  * CDK Assets will not be include in the iteration due to complexities around asset permissioning accross accounts and regions
+
+## Internal FAQ
+
+### Why are we doing this?
+
+Today, CDK does not support the creation and updating of CloudFormation stack sets directly.
+This solution would also enable deploying to existing stack sets with low effort.
+
+CDK does provide the `cdk.CfnStackSet` construct, which could accomplish similar behavior.
+The difference is that the `cdk.CfnStackSet` construct is a resource in a single CloudFormation stack with the resource type `AWS::CloudFormation::StackSet`,
+where stack provisions the actual stack set.
+This requires a 1:1 mapping of parameters and tags that need to be passed from the stack to the stack set.
+Customer can deploy to existing stack sets, but requires some migration steps.
+The customer would need to create a new stack,
+and import the stack set into the resource `AWS::CloudFormation::StackSet`.
+
+### Why should we _not_ do this?
+
+We might not want to do this since CDK does provide the `cdk.CfnStackSet` construct, which could accomplish similar behavior.
+The intermdiate stack may not be ideal for some customers, and may require large migration effort for customers who manage many stacksets.
+
+### What changes are required to enable this change?
+
+To enable this feature, we would need to introduce:
+
+* A `StackSet` construct for customers to extend
+* A new assembly/artifact type for the Cloud Assembly
+* A new stack set synthesizer
+  * A new stack set synthesizer is required to emit new stack set artifact and properties to the CloudAssembly
+* Deployment code that creates/updates stack sets using AWS CloudFormation APIs
+
+### Is this a breaking change?
+
+No.
+
+### What are the drawbacks of this solution?
+
+The handling of CDK assets is more complex than a normal stack CDK app.
+Stack set instances are deployed independently from CDK context, and exist in different accounts/regions.
+This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+
+An alternative is the usage of `cdk.CfnStackSet`, which is not ideal for customers with existing stack sets since there is significant migration effort.
+
+### What is the high level implementation plan?
+
+* Update default boostrap template to include default stack set roles
+* Create `StackSet` core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+* Add deployment code for creating/updating stack sets
+
+### Are there any open issues that need to be addressed later?
+
+* CDK asset handling
+* Future support for service managed stack sets
+* Future support for delete stack set
+* CDK Pipelines
+
+## Appendix
+
+### CDK Bootstrap CLI Changes
+
+To support stack sets,
```

Comment:
Repeat from above: can we use/expand the CDK pipelines roles we already have in the bootstrap stack? Intuitively these should suffice since CDK Pipelines is basically doing the same thing as stack sets.